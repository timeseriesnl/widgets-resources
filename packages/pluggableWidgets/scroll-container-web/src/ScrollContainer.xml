<?xml version="1.0" encoding="utf-8"?>
<widget id="com.mendix.widget.web.scrollcontainer.ScrollContainer" needsEntityContext="true" offlineCapable="true"
        pluginWidget="true"
        xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../../node_modules/mendix/custom_widget.xsd">
    <name>Scroll container</name>
    <description>Displays a scrollable container</description>
    <icon />
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="Width">
                <property key="widthType" type="enumeration" defaultValue="full">
                    <caption>Width</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="full">Full width</enumerationValue>
                        <enumerationValue key="pixels">Pixels</enumerationValue>
                        <enumerationValue key="percentage">Percentage</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="widthPixels" type="decimal" defaultValue="500">
                    <caption>Pixels</caption>
                    <description>The amount of pixels to be considered as width.</description>
                </property>
                <property key="widthPercentage" type="decimal" defaultValue="100">
                    <caption>Percentage</caption>
                    <description>The amount of percentage to be considered as width.</description>
                </property>
                <property key="content" type="widgets">
                    <caption>Content</caption>
                    <description />
                </property>
            </propertyGroup>
        </propertyGroup>
    </properties>
</widget>
