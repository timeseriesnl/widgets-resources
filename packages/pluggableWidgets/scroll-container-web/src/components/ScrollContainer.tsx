import { createElement, CSSProperties, PropsWithChildren, ReactElement } from "react";
import classNames from "classnames";

interface ScrollContainerProps {
    className: string;
    width?: number;
    percentage?: number;
    styles?: CSSProperties;
}

export function ScrollContainer(props: PropsWithChildren<ScrollContainerProps>): ReactElement {
    return (
        <div
            className={classNames("widget-scroll-container", props.className)}
            style={{
                ...(props.width
                    ? { width: `${props.width}px` }
                    : props.percentage
                    ? { width: `${props.percentage}%` }
                    : undefined),
                ...props.styles
            }}
        >
            <div className="widget-scroll-container-wrapper">{props.children}</div>
        </div>
    );
}
