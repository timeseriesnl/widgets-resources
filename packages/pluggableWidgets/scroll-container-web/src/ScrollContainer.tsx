import { createElement, ReactElement } from "react";
import { ScrollContainerContainerProps } from "../typings/ScrollContainerProps";
import { ScrollContainer as ScrollContainerComponent } from "./components/ScrollContainer";

import "./ui/scroll-container.scss";

export function ScrollContainer(props: ScrollContainerContainerProps): ReactElement {
    return (
        <ScrollContainerComponent
            className={props.class}
            percentage={props.widthType === "percentage" ? props.widthPercentage?.toNumber() : undefined}
            width={props.widthType === "pixels" ? props.widthPixels?.toNumber() : undefined}
            styles={props.style}
        >
            {props.content}
        </ScrollContainerComponent>
    );
}
