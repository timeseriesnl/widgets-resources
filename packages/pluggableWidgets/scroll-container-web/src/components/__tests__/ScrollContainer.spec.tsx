import { createElement } from "react";
import { shallow } from "enzyme";
import { ScrollContainer } from "../ScrollContainer";

describe("Scroll Container", () => {
    it("renders correctly the structure", () => {
        const scrollContainer = shallow(<ScrollContainer className="scroll-container-test" />);
        expect(scrollContainer).toMatchSnapshot();
    });

    describe("with fixed width", () => {
        it("renders correctly the structure with pixels", () => {
            const scrollContainer = shallow(<ScrollContainer className="scroll-container-pixels" width={500} />);
            expect(scrollContainer).toMatchSnapshot();
        });

        it("renders correctly the structure with percentage", () => {
            const scrollContainer = shallow(
                <ScrollContainer className="scroll-container-percentage" percentage={70} />
            );
            expect(scrollContainer).toMatchSnapshot();
        });
    });
});
