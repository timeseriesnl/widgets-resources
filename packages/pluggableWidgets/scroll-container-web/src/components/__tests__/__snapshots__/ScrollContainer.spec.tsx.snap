// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll Container renders correctly the structure 1`] = `
<div
  className="widget-scroll-container scroll-container-test"
  style={Object {}}
>
  <div
    className="widget-scroll-container-wrapper"
  />
</div>
`;

exports[`Scroll Container with fixed width renders correctly the structure with percentage 1`] = `
<div
  className="widget-scroll-container scroll-container-percentage"
  style={
    Object {
      "width": "70%",
    }
  }
>
  <div
    className="widget-scroll-container-wrapper"
  />
</div>
`;

exports[`Scroll Container with fixed width renders correctly the structure with pixels 1`] = `
<div
  className="widget-scroll-container scroll-container-pixels"
  style={
    Object {
      "width": "500px",
    }
  }
>
  <div
    className="widget-scroll-container-wrapper"
  />
</div>
`;
